inference-engine/src/transformations/include/ngraph_ops/nms_static_shape_ie.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include <algorithm>
+#include <string>
+
+#include <transformations_visibility.hpp>
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph {
+namespace op {
+namespace internal {
+
+template <typename BaseNmsOp>
+class NmsStaticShapeIE : public BaseNmsOp {
+public:
+    NGRAPH_RTTI_DECLARATION;
+
+    using Attributes = typename BaseNmsOp::Attributes;
+
+    /// \brief Constructs a NmsStaticShapeIE operation
+    ///
+    /// \param boxes Node producing the box coordinates
+    /// \param scores Node producing the box scores
+    /// \param attrs Attributes of the operation
+    NmsStaticShapeIE(const Output<Node>& boxes,
+                     const Output<Node>& scores,
+                     const Attributes& attrs) : BaseNmsOp(boxes, scores, attrs) {
+        validate_and_infer_types();
+    }
+    void validate_and_infer_types() override;
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<NmsStaticShapeIE>(new_args.at(0), new_args.at(1), this->m_attrs);
+    }
+};
+
+template <typename BaseNmsOp>
+void NmsStaticShapeIE<BaseNmsOp>::validate_and_infer_types() {
+    const auto boxes_ps = this->get_input_partial_shape(0);
```

Comment:
nitpick: `this->` is not needed

Reply:
If using template as base class the base function may be hidden, using this can resolve the problem. Reference: [Why do I have to access template base class members through the this pointer?](https://stackoverflow.com/questions/4643074/why-do-i-have-to-access-template-base-class-members-through-the-this-pointer)